samza-api/src/main/java/org/apache/samza/operators/windows/WindowFunction.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.operators.windows;
+
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.operators.data.MessageEnvelope;
+
+/**
+ * A {@link WindowFunction} transform slices a stream into smaller finite chunks for further processing. Programmers should
+ * use the API methods of {@link Windows} to specify their windowing functions.
+ *
+ * <p>There are the following aspects to windowing in Samza:
```

Comment:
Slightly prefer the `<p>` on the same line for readability instead of having a separate line just for a tag. Not a huge fan of one-line tags. Also, I ended up indenting the comment a bit so that it reads slightly better. (not that it matters when the html is rendered ;) )